include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1662,9 +1662,9 @@ __pattern_rotate(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Iterator
     auto __temp_rng_rw =
         oneapi::dpl::__ranges::all_view<_Tp, __par_backend_hetero::access_mode::read_write>(__temp_buf.get_buffer());
     oneapi::dpl::__par_backend_hetero::__parallel_for(_BackendTag{}, ::std::forward<_ExecutionPolicy>(__exec), __brick,
-                                                      __n, __temp_rng_rw, __buf.all_view());
+                                                      __n, __temp_rng_rw, __buf.all_view()).wait();
```

Comment:
I believe I agree with the clang-format suggestion to move `.wait()` to the next line, but it is your call.